batchflow/tests/dataformat_test.py

```diff
@@ -0,0 +1,187 @@
+# pylint: disable=missing-docstring, redefined-outer-name
+import os
+import warnings
+warnings.filterwarnings("ignore")
+os.environ['TF_CPP_MIN_LOG_LEVEL'] = '3'
+
+import pytest
+import numpy as np
+import tensorflow as tf
+
+from batchflow import Pipeline, ImagesBatch, Dataset
+from batchflow import B, V
+from batchflow.models.tf import VGG7, ResNet18, Inception_v1
+
+
+AVAILABLE_MODELS = [VGG7, ResNet18, Inception_v1]
+LOCATIONS = set(['initial_block', 'body', 'block', 'head'])
+
+
+@pytest.fixture()
+def model_setup():
+    """ Pytest fixture to generate fake dataset and model config with desired format.
+
+    Parameters
+    ----------
+    d_f: {'channels_last', 'channels_first'}
+        desired format of returns.
+
+    Returns
+    -------
+    tuple
+        First element is instance of Dataset.
+        Second element is dict with model description.
+    """
+    def _model_setup(d_f):
+        if d_f == 'channels_last':
+            shape_in = (100, 100, 2)
+        elif d_f == 'channels_first':
+            shape_in = (2, 100, 100)
+
+        size = 50
+        batch_shape = (size,) + shape_in
+        images_array = np.random.random(batch_shape)
+        labels_array = np.random.choice(10, size=size)
+        data = images_array, labels_array
+        fake_dataset = Dataset(index=size,
+                               batch_class=ImagesBatch,
+                               preloaded=data)
+
+        model_config = {'inputs': {'images': {'shape': shape_in},
+                                   'labels': {'classes': 10}},
+                        'initial_block/inputs': 'images'}
+        return fake_dataset, model_config
+
+    return _model_setup
+
+@pytest.fixture()
+def get_model_pipeline():
+    """ Creates instance of Pipeline that is configured to use given model
+    with passed parameters.
+
+    Parameters
+    ----------
+
+    model_class : subclass of TFModel
+        Architecture of model. List of available models is defined at 'AVAILABLE_MODELS'.
+
+    current_config : dict
+        Dictionary with parameters of model.
+
+    Returns
+    -------
+    Pipeline
+        Test pipeline that consists of initialization of model and
+        preparing for training with given config.
+    """
+    def _get_model_pipeline(model_class, current_config):
+        test_pipeline = (Pipeline()
+                         .init_variable('current_loss')
+                         .init_model('dynamic', model_class,
+                                     'TestModel', current_config)
+                         .to_array()
+                         .train_model('TestModel',
+                                      fetches='loss',
+                                      images=B('images'),
+                                      labels=B('labels'),
+                                      save_to=V('current_loss'))
+                         )
+        return test_pipeline
+    return _get_model_pipeline
```

Comment:
You don't need a complicated fixture here. You can just return `Pipeline` and within a test function apply a config to it through `set_config(pipeline_config)`. For that to work, `pipeline_config` should contain keys `model_class` and `model_config`.